### DISCUSSION TITLE
FDS Runtime

### DISCUSSION THREAD

**Discussion by Tbuttts21** [firemodels/fds#12157] [General]
Just wondering if anyone has done any benchmarking on factors that impact FDS runtime. I was going to start a bunch of simulations to see how different settings (number of meshes, number of cells, wind speed, domain size, etc.) impact how fast the simulations run. I figured I would ask here first in case others have already run these evaluations. 
Thanks!

**Comment by drjfloyd** [comment]
Roughly speaking:

-Keeping the same number of cells, run time will scale linearly up to a point with the number of meshes. Non-linear effects will be seen that are dependant on the your specific hardware (CPU, memory bus, network bandwidth/latency, etc.). This is something you would need to figure out by running a scaling test on your hardware (this is discussed in the User's Guide and the inputs are available from GitHub).
-For the same grid size and number of meshes, more cells (i.e. larger domain) would scale linearly with the number of cells up to the point where you hit limits based on available RAM.
-For the same domain and number of meshes, more cells (i.e. smaller grid size) will scale non-linearly with the number of cells.  The time step size is limted by the CFL limit, i.e., the maximum time-step scales with grid size.
-Wind speed: depends. The global timestep is limited by whichever grid cell has the highest CFL number. If the highest velocity is not due to the wind, then changing the wind speed should not have a large impact.
